lib/cdo/rack/request.rb

```diff
@@ -101,6 +102,15 @@ def user_id_from_session_cookie
     rescue
       return nil
     end
+
+    def country
```

Comment:
Just confirming: do both the cloudfront country and the fallback country work for Pegasus requests too?  And country is now part of the cache variant for all sites?

Reply:
> do both the cloudfront country and the fallback country work for Pegasus requests too? And country is now part of the cache variant for all sites?

No, this won't work for [all] Pegasus requests as it's currently configured, because country isn't added to the cache variant for cached pages. I've only added the country header to _non-cached_ CloudFront path behaviors for now to avoid impact on any existing cache behavior.

If we do end up with cases where we want to vary _cached_ page-responses based on country-code (in addition to the language-varying we already do on these pages), we can change the cache configuration to make that work as well. However, such a change would multiply the number of uniquely-cached pages by the number of countries (so the total variants for each cached page will be languages * countries * user_types). This would probably be fine, but would be significant enough cache change that we'd want to watch for any increased load when deploying.

Reply:
As a followup, we now do vary the pegasus caching by country courtesy of https://github.com/code-dot-org/code-dot-org/pull/26753.